coredhcp.go

```diff
@@ -104,20 +104,26 @@ func (s *Server) LoadPlugins(conf *config.Config) ([]*plugins.Plugin, []*plugins
 // It will not reply if the resulting response is `nil`.
 func (s *Server) MainHandler6(conn net.PacketConn, peer net.Addr, req dhcpv6.DHCPv6) {
 	var (
-		resp, tmp dhcpv6.DHCPv6
-		stop      bool
-		err       error
+		resp dhcpv6.DHCPv6
+		stop bool
+		err  error
 	)
 
 	// Create a suitable basic response packet
-	switch req.Type() {
+	tmp := req
+	switch tmp.Type() {
 	case dhcpv6.MessageTypeSolicit:
-		tmp, err = dhcpv6.NewAdvertiseFromSolicit(req.(*dhcpv6.Message))
+		m := tmp.(*dhcpv6.Message)
+		if m.GetOneOption(dhcpv6.OptionRapidCommit) != nil {
+			tmp, err = dhcpv6.NewReplyFromMessage(m)
+		} else {
+			tmp, err = dhcpv6.NewAdvertiseFromSolicit(req.(*dhcpv6.Message))
```

Comment:
```
tmp, err = dhcpv6.NewAdvertiseFromSolicit(req.(*dhcpv6.Message))
```

can be changed to:

```
tmp, err = dhcpv6.NewAdvertiseFromSolicit(m)
```

It will be more consistent with what is done for the rapid commit

Reply:
good catch